app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -1,15 +1,84 @@
 package com.lightricks.feedexercise.data
 
+import android.content.Context
+import android.util.Log
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.MockFeedApiService
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
 import org.junit.runner.RunWith
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
+
 @RunWith(AndroidJUnit4::class)
 class FeedRepositoryTest {
-   //todo: add the tests here
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var db: FeedDatabase
+    private lateinit var feedApiService: FeedApiService
+
+    @Before
+    fun setup() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        createDb(context)
+        createApiService(context)
+    }
+
+    private fun createDb(context: Context) {
+        db = Room.inMemoryDatabaseBuilder(
+            context, FeedDatabase::class.java
+        ).build()
+    }
+
+    private fun createApiService(context: Context) {
+        feedApiService = MockFeedApiService(context)
+    }
+
+    @Test
+    fun refresh_whenFetchDataFromNetwork_saveFeedToDatabase() {
+        val feedRepository = FeedRepository(feedApiService, db)
+
+        feedRepository.feedItems.blockingObserve()
+        assertThat(feedRepository.feedItems.value?.size).isEqualTo(0)
+        feedRepository.refresh().test().await()
+        feedRepository.feedItems.blockingObserve()
+        assertThat(feedRepository.feedItems.value?.size).isEqualTo(db.feedDao().count())
```

Comment:
What of the count is still zero?

Reply:
I'm not sure that I understand why it can happen. I assumed that _count()_ won't be zero, since the json is not empty.

After re-reading the exercise, I wonder if this test was supposed to check the Db, and not the feedItems:
_assertThat(db.getAll().value).containsExactly(*expectedListValues)_
Am I right?

Reply:
You are comparing the size of feed items with database rows count. But there is no guarantee that the database actually contains any items. You assume that `refresh()` has put some items into the database, but this is a side effect. It would be better to insert the items into the database directly using `FeedDao` and then verify that `feedItems` contains them (or at least is not empty).

Reply:
Imagine this scenario: you call `feedRepository.refresh()`, the repository calls the API service, a network error occurs, nothing is inserted into the database. 

Now you compare `feedItems.value?.size` with `db.feedDao().count()` and they are both 0. So, the test passes, but it didn't actually test reading feed items from the database.

Reply:
One test is supposed to test `refresh()`, the other one is supposed to test `feadItems`.

Reply:
Sorry, I got confused about what test we are talking about :) 
Since in this test you are testing `refresh()` then the algorithm should be:
1. Check that the DB is empty
2. Call `feedRepository.refresh()`
3. Verify that DB contains the expected items (directly, not via `feedItems`)